### PR TITLE
Release/1.2.8

### LINKED ISSUE #456: Release version 1.2.8
This issue is for tracking the 1.2.8 release. Target release date: **August 19 2024**

## Pre-release steps

- [x] Merge https://github.com/woocommerce/woocommerce-bookings/pull/3831  
- [x] Merge https://github.com/woocommerce/woocommerce-bookings/pull/3842 
- [x] Merge #425   
- [x] Merge #447   
- [x] Merge #451  
- [x] Merge #452  
- [x] Merge #454 
- [x] Merge #455              

## Release steps

- [x] Branch: Starting from `trunk`, cut a release branch named `release/1.2.8`
- [x] Version bump: Bump the version number in `woocommerce-accommodation-bookings.php`, `readme.txt`, `package.json` and `package-lock.json` to 1.2.8. In `woocommerce-accommodation-bookings.php` update both the plugin "Version:" property and the plugin `WC_ACCOMMODATION_BOOKINGS_VERSION` constant
- [x] Update `@since`: Find all new `@since x.x.x` / `@since n.e.x.t` lines and update those with the new version number, 1.2.8
- [x] Generate changelog: Locally, generate the changelog by running the following Woorelease command: `php woorelease.phar cl:generate --product_version=1.2.8 https://github.com/woocommerce/woocommerce-accommodation-bookings/tree/trunk`. Ensure the changelog updates look correct
- [x] Update changelog: Take the changelog generated from the previous step and update the `changelog.txt` and `readme.txt` in the release PR with those changes 
- [x] Open PR: Open a new PR against `trunk` from the release branch, `release/1.2.8`
- [x] Run tests: On this release PR, add the label `needs: qit default tests`. This will trigger the default QIT tests (activation, API and E2E). Ensure these tests pass as well as all other applicable tests
- [x] Merge: Merge the release PR into `trunk`
- [x] Prepare ZIP: Prepare the zip by running Woorelease locally on the `trunk` branch: `php woorelease.phar build https://github.com/woocommerce/woocommerce-accommodation-bookings/tree/trunk`. Make sure you are running the proper versions of `node`, `npm` and `composer` to ensure the built files are consistent across releases. Versions can be found in [`package.json`](https://github.com/woocommerce/woocommerce-accommodation-bookings/blob/trunk/package.json)
- [x] Compare: Download the latest [released zip](https://github.com/woocommerce/woocommerce-accommodation-bookings/releases) and then run a diff compare against the generated zip, ensuring all expected changes are there (and no unexpected changes)
- [x] Test ZIP: Take the ZIP generated from the previous step and test this locally, ensuring version numbers look correct, the changelog looks correct and the plugin installs and activates properly
- [x] Attach ZIP: Attach the ZIP to this issue and ping the Automattic team in the `#a8c-10up-woo-support` channel to complete the release

### PR DESCRIPTION
**Pull request by dkotter**
### Changes proposed in this Pull Request:

Version bump and prep for the 1.2.8 release

Closes #456 

### Steps to test the changes in this Pull Request:

Verify all file changes look accurate

### CODE REVIEW

**Comment by dkotter** [comment]
E2E tests are failing here. Will need to look at those before we proceed

**Review by vikrampm1 — APPROVED** [review]

**Comment by axi** [comment]
`switchTab()` method search for the **last** `'.wc-tabs > li > a'` with name "Availability" but it's now the `#bookings_availability` that is listed last in the code (and hidden) so test script cannot click it.

```
<li class="accommodation_bookings_availability_options accommodation_bookings_availability_tab show_if_accommodation-booking accommodation_bookings_tab bookings_availability_tab advanced_options">
    <a href="#accommodation_bookings_availability]"><span>Availability</span></a>
</li>
<li class="bookings_resources_options bookings_resources_tab show_if_booking">
    <a href="#bookings_resources"><span>Resources</span></a>
</li>
<li class="bookings_availability_options bookings_availability_tab show_if_booking">
    <a href="#bookings_availability"><span>Availability</span></a>
</li>
```

Modifying the selector looks to work : `'.wc-tabs > li > a' => '.wc-tabs > li:visible > a'`
in `tests/e2e/utils/index.js`
```javascript
/**
 * Switch tab in add/edit product page.
 *
 * @param {Page}   page          Playwright page object
 * @param {string} tabName       Tab name
 * @param {string} panelSelector Options Panel selector
 */
export async function switchTab(page, tabName, panelSelector = false) {
	await page
		.locator('.wc-tabs > li:visible > a', { hasText: tabName })
		.last()
		.click();
	if (panelSelector) {
		await expect(page.locator(panelSelector)).toBeVisible();
	}
}
```

**Comment by axi** [comment]
E2E tests went through on my machine with the last commit

**Comment by iamdharmesh** [comment]
Thanks a lot for the [suggestion](https://github.com/woocommerce/woocommerce-accommodation-bookings/pull/457#issuecomment-2305309451) @axi. It worked.

> E2E tests went through on my machine with the last commit

Yes, it was working fine for me as well. I have fixed it now. Thanks. 

@dkotter we are ready for the next steps here.

File: changelog.txt [hunk 1]
```diff
@@ -1,5 +1,13 @@
 *** Changelog ***
 
+= 1.2.8 - 2024-08-19 =
```
Comment (dkotter) [inline changelog.txt]:
```suggestion
= 1.2.8 - 2024-08-28 =
```

File: readme.txt [hunk 2]
```diff
@@ -38,6 +38,14 @@ If the prices shown on the product do not match the prices defined in the dashbo
 
 == Changelog ==
 
+= 1.2.8 - 2024-08-19 =
```
Comment (dkotter) [inline readme.txt]:
```suggestion
= 1.2.8 - 2024-08-28 =
```

**Pull request merged**